src/plugins/discover/public/application/components/sidebar/lib/field_calculator.test.ts

```diff
@@ -187,48 +173,96 @@ describe('fieldCalculator', function () {
   });
 
   describe('getFieldValueCounts', function () {
-    let params: { hits: any; field: any; count: number; indexPattern: IndexPattern };
+    let params: FieldValueCountsParams;
     beforeEach(function () {
       params = {
         hits: _.cloneDeep(realHits),
-        field: indexPattern.fields.getByName('extension'),
+        field: indexPattern.fields.getByName('extension') as IndexPatternField,
         count: 3,
         indexPattern,
       };
     });
 
+    it('counts the top 5 values by default', function () {
+      params.hits = params.hits.map((hit: Record<string, any>, i) => ({
+        ...hit,
+        _source: {
+          extension: `${hit._source.extension}-${i}`,
+        },
+      }));
+      params.count = undefined;
+      const extensions = getFieldValueCounts(params);
+      expect(extensions).toBeInstanceOf(Object);
+      expect(extensions.buckets).toBeInstanceOf(Array);
+      const buckets = extensions.buckets as Bucket[];
+      expect(buckets.length).toBe(5);
+      expect(extensions.error).toBe(undefined);
+    });
+
+    it('counts only distinct values if less than default', function () {
+      params.count = undefined;
+      const extensions = getFieldValueCounts(params);
+      expect(extensions).toBeInstanceOf(Object);
+      expect(extensions.buckets).toBeInstanceOf(Array);
+      const buckets = extensions.buckets as Bucket[];
+      expect(buckets.length).toBe(4);
+      expect(extensions.error).toBe(undefined);
+    });
+
+    it('counts only distinct values if less than specified count', function () {
+      params.count = 10;
+      const extensions = getFieldValueCounts(params);
+      expect(extensions).toBeInstanceOf(Object);
+      expect(extensions.buckets).toBeInstanceOf(Array);
+      const buckets = extensions.buckets as Bucket[];
+      expect(buckets.length).toBe(4);
+      expect(extensions.error).toBe(undefined);
+    });
+
     it('counts the top 3 values', function () {
-      const extensions = fieldCalculator.getFieldValueCounts(params);
+      const extensions = getFieldValueCounts(params);
       expect(extensions).toBeInstanceOf(Object);
       expect(extensions.buckets).toBeInstanceOf(Array);
-      expect(extensions.buckets.length).toBe(3);
-      expect(_.map(extensions.buckets, 'value')).toEqual(['html', 'php', 'gif']);
+      const buckets = extensions.buckets as Bucket[];
+      expect(buckets.length).toBe(3);
+      expect(_.map(buckets, 'value')).toEqual(['html', 'gif', 'php']);
       expect(extensions.error).toBe(undefined);
     });
 
     it('fails to analyze geo and attachment types', function () {
-      params.field = indexPattern.fields.getByName('point');
-      expect(fieldCalculator.getFieldValueCounts(params).error).not.toBe(undefined);
+      params.field = indexPattern.fields.getByName('point') as IndexPatternField;
+      expect(getFieldValueCounts(params).error).not.toBe(undefined);
```

Comment:
i think toBe(undefined) here can also be replaced by toBeUndefined()? i am not sure which syntax we prefer tho.

Reply:
Thanks, nice catch!